### DISCUSSION TITLE
Contribute a new language with existing grammar

### DISCUSSION THREAD

**Discussion by BoykoAlex** [microsoft/vscode-discussions#598] [Extension Development QnA] [answered]
I'd like to define a new language, i.e. `spring-boot-properties-yaml` and re-use the built-in yaml grammar for the language. Thus the language would be defined as YAML grammar with file association `application-*.yml` files.
Another instance of the same issue is `spring-boot-properties` language which would re-use `java-properties` grammar defined in `vscode-java` extension.

Currently I'm forced to contribute the grammar for the new language although in both cases the grammar already exists and I'd rather re-use the existing one than copy-paste in my extension.

Any advice on how to re-use already defined grammar for the new language contribution?

**Comment by isidorn** [comment]
I think this is not possible. @alexr00 can confirm.

**Comment by alexr00** [comment, marked as answer]
I agree, I don't think we support this.
